### PR TITLE
TypeName annotations added

### PR DESCRIPTION
**Pull request by sormuras**
#136

### CODE REVIEW

**Comment by swankjesse** [comment]
Love it. LGTM

**Comment by swankjesse** [comment]
(usual flattening request applies)

**Comment by sormuras** [comment]
Applied.

And love it more than the former solution, too.

**Comment by sormuras** [comment]
`TypeName` needs some more love... wait a sec.

**Comment by sormuras** [comment]
Done.

File: src/main/java/com/squareup/javapoet/TypeName.java [hunk 1]
```diff
@@ -87,14 +88,34 @@
...
   // Package-private constructor to prevent third-party subclasses.
-  TypeName() {
-    this(null);
+  TypeName(List<AnnotationSpec> annotations) {
+    this(null, annotations);
+  }
+
+  public TypeName annotated(AnnotationSpec... annotations) {
+    Util.checkNotNull(annotations, "annotations == null");
+    return new TypeName(keyword, Arrays.asList(annotations));
```
Comment (sormuras) [inline src/main/java/com/squareup/javapoet/TypeName.java]: This replaces any existing annotations with the new collection. If the user wants to merge the old and new annotations, it has to be done before calling this method. A nice side-effect of this solution, that an empty new collection strips any existing annotations.

**Pull request merged**